### PR TITLE
feat: #3530 Add support for default choice in react dropdown

### LINKED ISSUE #3530: Allow a user to specify a default shown in a dropdown. (sprk-react)
## Description

Currently, text that, by default, reads 'Make a selection' is rendered in the dropdown on load. We'd like to allow setting default: true on a supplied choice, that would let one of the choices be the rendered one on load.

See [React Informational Dropdown](https://react.sparkdesignsystem.com/?path=/docs/components-dropdown--informational) for existing component use. 

Default render today: 

Screen Shot 2020-10-02 at 1 19 24 PM

Default render with default option supplied:

Screen Shot 2020-10-02 at 1 19 35 PM

## Acceptance Criteria

- The choice object, as used in the [component](https://github.com/sparkdesignsystem/spark-design-system/blob/staging/react/src/components/dropdown/SprkDropdown.js), should allow a new property: 'default'.
- The [component](https://github.com/sparkdesignsystem/spark-design-system/blob/staging/react/src/components/dropdown/SprkDropdown.js) will need to make use of the new property and render the default choice correctly. If there is no default choice, the component should continue to render the text that reads 'Make a selection' by default.
- There should be additional [unit tests](https://github.com/sparkdesignsystem/spark-design-system/blob/staging/react/src/components/dropdown/SprkDropdown.test.js) to cover this case, and they should pass.

### PR DESCRIPTION
**Pull request by sanchitgupta001**
## What does this PR do?
Adds handling to select default choice in react dropdown, if isDefault: true is provided in option config.

### Associated Issue
Fixes #3530

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR then please remove it.

### Documentation
 - [ ] Update Spark Docs

### Code
 - [ ] Build Component in HTML
 - [ ] Build Component in Angular
 - [x] Build Component in React
 - [ ] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)
 - [ ] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [ ] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [ ] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [ ] Apple Safari
  - [ ] Apple Safari (Mobile)
  
### Accessibility Testing
  - [ ] Axe browser extension
  - [ ] Jaws Inspect
  - [ ] VoiceOver (iOS) or Talkback (Android)

### Deploy Preview Reviewed and Approved
 - [ ] Design Team
 - [ ] Accessibility Team

### Screenshots
Add screenshots to help explain your PR if you'd like. However, this is not
expected.

Screenshot 2020-10-03 at 1 27 37 AM
Screenshot 2020-10-03 at 1 27 44 AM

### CODE REVIEW

**Comment by RCopeland** [comment]
Hey @sanchitgupta001, thanks for opening this! I'll be reviewing it in the next few days. Thanks!

File: react/src/components/dropdown/SprkDropdown.js [hunk 1]
```diff
@@ -282,8 +304,14 @@ SprkDropdown.propTypes = {
         href: PropTypes.string,
         /** The text inside the choice item. */
         text: PropTypes.string,
+        /**
+         * If set to true, current option will be set as default selected option
+         *  */
+        default: PropTypes.bool,
```
Comment (afebbraro) [inline react/src/components/dropdown/SprkDropdown.js:310]: Hey! Thanks for the PR. Could you change this to isDefault?

Comment (sanchitgupta001) [reply react/src/components/dropdown/SprkDropdown.js]: have updated the pull request with asked changes

File: react/src/components/dropdown/SprkDropdown.test.js [hunk 2]
```diff
@@ -25,30 +25,48 @@ describe('SprkDropdown:', () => {
     expect(wrapper.find('.sprk-c-Dropdown__footer').length).toBe(1);
   });
 
-  it('should add classes to the dropdown when additionalClasses has a value', () => {
```
Comment (RCopeland) [inline react/src/components/dropdown/SprkDropdown.test.js:28]: can you revert the style changes to the test file and leave only the added tests?

**Review by RCopeland — CHANGES_REQUESTED** [review]

File: react/src/components/dropdown/SprkDropdown.test.js (hunk 2 above)
Comment (sanchitgupta001) [reply react/src/components/dropdown/SprkDropdown.test.js:28]: had done these changes because eslint was breaking as a part of git pre-commit  hook

Comment (RCopeland) [reply react/src/components/dropdown/SprkDropdown.test.js:28]: Ahh ok, I'll take a look

Comment (RCopeland) [reply react/src/components/dropdown/SprkDropdown.test.js:28]: You're right! Thanks for fixing those line lengths as part of this task!

**Review by RCopeland — APPROVED** [review]

**Pull request merged**